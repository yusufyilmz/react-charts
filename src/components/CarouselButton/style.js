import styled from 'styled-components';


export const Div = styled.div`
    height: 50px;
    width: 50px;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 50%;
    cursor: pointer;
    transition: transform ease-in .1s;
    top: 70%;
    z-index: 999;
    color: #fff;
    position: absolute;
    left: ${props =>  props.type === 'left' && '5px'};
    right: ${props =>  props.type === 'right' && '5px'};
    &hover {
        transition: transform ease-in .1s;
        transform: scale(1.1)
    }
`;

Div.displayName = 'div'

export const Icon = styled.i`
    color: black;
    font-size: xx-large;
`;

Icon.displayName = 'i'
