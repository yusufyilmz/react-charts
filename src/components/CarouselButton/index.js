

import React from 'react';
import {Div, Icon} from './style';


const CarouselButton = ({ type, onClick }) => {

    const getIconClassName = () => {
        return `fa fa-angle-double-${type}`
    }

    return (
        <Div type={type} onClick={onClick}>
             <Icon className={getIconClassName()}></Icon>
        </Div>
    )
}

export default CarouselButton
