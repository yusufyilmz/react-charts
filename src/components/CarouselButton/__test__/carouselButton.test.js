

import React from 'react';
import CarouselButton from '../index';
import { shallow } from 'enzyme'

describe('CAROUSELBUTTON COMPONENT', () => {

    let props, wrapper;

    beforeEach(() => {
        props = {
            type: 'left',
            onClick: jest.fn()
        }
        wrapper = shallow(
            <CarouselButton {...props} />
        );
    });

    test('renders element correctly', () => {
        expect(wrapper.find('div').length).toBe(1)

    });

    test('renders classname correctly', () => {
        expect(wrapper.find('i').hasClass(`fa fa-angle-double-${props.type}`)).toBeTruthy()
    });

    test('renders onclick correctly', () => {
        wrapper.find('div').simulate('click');
        expect(props.onClick.mock.calls.length).toEqual(1);
    });

    // test('snapchat correctly', () => {
    //     expect(wrapper).toMatchSnapshot();
    // });

})
