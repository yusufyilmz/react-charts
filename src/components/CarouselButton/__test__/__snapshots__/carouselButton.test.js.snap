// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CAROUSELBUTTON COMPONENT snapchat correctly 1`] = `
<div
  onClick={[MockFunction]}
  type="left"
>
  <i
    className="fa fa-angle-double-left"
  />
</div>
`;
